### PR TITLE
Add support for timezone

### PR DESCRIPTION
**Pull request by Aiee**
As title.

### CODE REVIEW

File: result_set.go [hunk 1]
```diff
@@ -722,15 +745,346 @@ func (p1 *PathWrapper) IsEqualTo(p2 *PathWrapper) bool {
...
+		localTime.Hour(),
+		localTime.Minute(),
+		localTime.Second(),
+		localTime.Nanosecond()/1000), nil
+}
+
+// getLocalTimeWithCostomOffset returns a string of local time using user specified offset.
+// Offset is in seconds.
+// Output format: HH:MM:MSMSMS
+func (t TimeWrapper) getLocalTimeWithTimezonOffset(timezoneOffsetSeconds int32) (string, error) {
```
Comment (laura-ding) [inline result_set.go:825]: `getLocalTimeWithTimezonOffset` - > `getLocalTimeWithTimezoneOffset`

File: result_set.go [hunk 2]
```diff
@@ -722,15 +745,346 @@ func (p1 *PathWrapper) IsEqualTo(p2 *PathWrapper) bool {
...
+		localTime.Hour(),
+		localTime.Minute(),
+		localTime.Second(),
+		localTime.Nanosecond()/1000), nil
+}
+
+// GetLocalTimeWithTimezonName returns  a string of local time using user specified timezone name.
+// timezoneNames are defined in IANA time zone database.
+// Output format: HH:MM:MSMSMS
+func (t TimeWrapper) GetLocalTimeWithTimezonName(timezoneName string) (string, error) {
```
Comment (laura-ding) [inline result_set.go:852]: `GetLocalTimeWithTimezonName` to `GetLocalTimeWithTimezoneName`

File: result_set.go [hunk 3]
```diff
@@ -722,15 +745,346 @@ func (p1 *PathWrapper) IsEqualTo(p2 *PathWrapper) bool {
...
+		localDateTime.Hour(),
+		localDateTime.Minute(),
+		localDateTime.Second(),
+		localDateTime.Nanosecond()/1000), nil
+}
+
+// getLocalDateTimeWithTimezonOffset returns a string of local time using user specified timezone offset.
+// Offset is in seconds.
+// Output format: yyyy-mm-ddTHH:MM:SS.MSMSMS
+func (dt DateTimeWrapper) getLocalDateTimeWithTimezonOffset(timezoneOffsetSeconds int32) (string, error) {
```
Comment (laura-ding) [inline result_set.go:1022]: `getLocalDateTimeWithTimezonOffset` -> `getLocalDateTimeWithTimezoneOffset`

File: result_set.go [hunk 4]
```diff
@@ -722,15 +745,346 @@ func (p1 *PathWrapper) IsEqualTo(p2 *PathWrapper) bool {
...
+		localDateTime.Hour(),
+		localDateTime.Minute(),
+		localDateTime.Second(),
+		localDateTime.Nanosecond()/1000), nil
+}
+
+// GetLocalDateTimeWithTimezonName returns a string of local time using user specified timezone name.
+// timezoneNames are defined in IANA time zone database.
+// Output format: yyyy-mm-ddTHH:MM:MSMSMS
+func (dt DateTimeWrapper) GetLocalDateTimeWithTimezonName(timezoneName string) (string, error) {
```
Comment (laura-ding) [inline result_set.go:1052]: `GetLocalDateTimeWithTimezonName` -> `GetLocalDateTimeWithTimezoneName`

File: result_set.go [hunk 5]
```diff
@@ -722,15 +745,346 @@ func (p1 *PathWrapper) IsEqualTo(p2 *PathWrapper) bool {
...
+		localDateTime.Day(),
+		localDateTime.Hour(),
+		localDateTime.Minute(),
+		localDateTime.Second(),
+		localDateTime.Nanosecond()/1000), nil
+}
+
+// getLocalDateTimeWithTimezonOffset returns a string of local time using user specified timezone offset.
+// Offset is in seconds.
+// Output format: yyyy-mm-ddTHH:MM:SS.MSMSMS
```
Comment (laura-ding) [inline result_set.go:1021]: return DateTime, not DateTimeStr

File: result_set.go [hunk 6]
```diff
@@ -130,18 +151,18 @@ func genNode(vertex *nebula.Vertex) (*Node, error) {
 	}, nil
```
Comment (laura-ding) [inline result_set.go:151]: need set timezone and add tests

File: result_set.go [hunk 7]
```diff
@@ -130,18 +151,18 @@ func genNode(vertex *nebula.Vertex) (*Node, error) {
...
 }
 
-func genRelationship(edge *nebula.Edge) (*Relationship, error) {
+func genRelationship(edge *nebula.Edge, timezoneInfo timezoneInfo) (*Relationship, error) {
 	if edge == nil {
-		return nil, fmt.Errorf("Failed to generate Relationship: invalid edge")
+		return nil, fmt.Errorf("failed to generate Relationship: invalid edge")
 	}
 	return &Relationship{
```
Comment (laura-ding) [inline result_set.go:158]: ditto

File: result_set.go (hunk 6 above)
Comment (Aiee) [reply result_set.go:151]: This senatio is tested in the `TestServiceDataIO` of `client_test.go` at line 257

**Review by laura-ding — APPROVED** [review]

**Pull request merged**